### PR TITLE
removed redundent free stmts

### PR DESCRIPTION
**Pull request by Atul-source**
Basically some free stmts are used even after **src_key** become null. In this PR I have addressed this issue

I have done testing of datapath and loading for  both ingress and egress on my local setup.

### CODE REVIEW

**Review by sanfern — APPROVED** [review]

File: traffic-mirroring/mirroring.c [hunk 1]
```diff
@@ -959,13 +958,11 @@ int main(int argc, char **argv)
                             "address %s",
                             dst_addr[i].addr);
                     ret = EXIT_FAILURE;
-                    free(dst_key);
                 }
```
Comment (aka320) [inline traffic-mirroring/mirroring.c:961]: Here we will end-up updating the Map eventhough this Conversion fails. Please check this error case if we need to return here before updating the Map.

File: traffic-mirroring/mirroring.c [hunk 2]
```diff
@@ -925,7 +925,6 @@ int main(int argc, char **argv)
                     log_err("Failed to update source endpoint bpf map");
```
Comment (aka320) [inline traffic-mirroring/mirroring.c:925]: Better to free malloc() memory before returning EXIT_FAILURE

**Review by aka320 — CHANGES_REQUESTED** [review]

File: traffic-mirroring/mirroring.c [hunk 3]
```diff
@@ -958,14 +958,13 @@ int main(int argc, char **argv)
...
-                    ret = EXIT_FAILURE;
-                    free(dst_key);
+		    free(dst_key);
+                    return (EXIT_FAILURE);
                 }
                 if (bpf_map_update_elem(dst_fd, dst_key, &dst_val, 0) < 0) {
                     log_err("Failed to update destination endpoint bpf map");
                     perror("ERROR: bpf_map_update_elem");
                     ret = EXIT_FAILURE;
-                    free(dst_key);
```
Comment (aka320) [inline traffic-mirroring/mirroring.c:968]: Why are we not returning here, in case of EXIT_FAILURE ?

Comment (Atul-source) [reply traffic-mirroring/mirroring.c:968]: Done and tested

**Review by aka320 — APPROVED** [review]
LGTM

**Pull request merged**